consumer/writer.go

```diff
@@ -19,20 +24,191 @@ func NewWriter(dir, templatesDir string) WriterConsumer {
 	}
 }
 
+func ensureDirectoryExists(path string) error {
+	err := os.MkdirAll(path, os.ModeDir)
+	if err != nil && err != os.ErrExist {
+		return err
+	}
+
+	return nil
+}
+
+func (wc *WriterConsumer) initTemplates() error {
+	if wc.template != nil {
+		return nil
+	}
+	// TODO check file ext
+	template, err := template.ParseGlob(path.Join(wc.templatesDir, "*.tmpl"))
```

Comment:
How does this work exactly? Will it load multiple files (with `tmpl` file extension) as one template? In any case, it should not be called `initTemplates` (plural) if it sets attribute called `template` (singular).

Reply:
Yes, it will load multiple files as a single template. The `template` package allows defining multiple templates to be defined in a single file e.g.:
```
{{define "T1"}}one{{end}}
{{define "T2"}}two{{end}}
```

Reply:
Now that I'm thinking about it, there should be some information about which part of the template is used with VM, storage, IP etc.

Reply:
I wouldn't worry about that since only one data type (VM) will use a template. I would even say that there is probably no need to read all the template files in the directory but specify just one template file for VMs (or APEL or whatever).